front/src/common/Map/Settings/MapSettingsSpeedLimits.tsx

```diff
@@ -44,6 +45,9 @@ const FormatSwitch = ({ name, icon: IconComponent, color = '', disabled }: Forma
       skip: !infraID,
     }
   );
+  const { data: speedLimitTags } = osrdEditoastApi.endpoints.getSpeedLimitTags.useQuery();
+
+  const allSpeedLimitTags = uniq(concat(speedLimitTags, speedLimitsTagsByInfraId));
```

Comment:
Maybe you could rather reuse the logic you did in `useStoreDataForSpeedLimitByTagSelector` with the concatenated list of tags. In the hook, you can also expose the `error` and `isError` from tagsByInfraId to use them here.

```suggestion
  const { speedLimitsByTags, speedLimitByTagsByInfraIdError, isErrorSpeedLimitByTagsByInfraId } = useStoreDataForSpeedLimitByTagSelector();
```

Reply:
This hook cannot be reused here, it calls up properties of the store that are outside the scope of the map